### PR TITLE
Add link validator and Linting 

### LINKED ISSUE #18: Add Link Validator Automation
There should be a CI job that checks for dead links within markdown files as discussed in [this closed PR](https://github.com/SovereignCloudStack/docs/pull/17) 

Example: https://github.com/gaurav-nelson/github-action-markdown-link-check

### PR DESCRIPTION
**Pull request by maxwolfs**
closes #18 

This PR is 

* adding a link validator action that checks on every PR if there are broken links within markdown files.
* enforcing pre commit checking and linting of markdown files using markdownlint with prettier, husky and lint-staged
* adding a markdownlint action that checks on every PR if all committed markdown comply to the markdown rules

The [guide](https://github.com/SovereignCloudStack/docs/blob/feat/add-link-validator/community/contribute/linting-guide.md) i still a first shot and needs further explanation how to set it up easily on other repositories

### CODE REVIEW

**Comment by maxwolfs** [comment]
I will integrate markdownlint action directly...

**Comment by maxwolfs** [comment]
@joshmue: I have extracted the comments from the json into a table

**Review by fkr — APPROVED** [review]
LGTM from a high-level look.

File: community/contribute/adding-docs-guide.md [hunk 1]
```diff
@@ -23,34 +23,37 @@ Your repository containing the documentation has to...
...
+  - [markdown file structure guideline](https://github.com/SovereignCloudStack/docs/blob/main/community/contribute/doc-files-structure-guide-md)
   - linting Rules
-  - [styleguide](../contribute/styleguide.md)
+  - [styleguide](https://github.com/SovereignCloudStack/docs/blob/main/community/contribute/styleguide.md)
 
 ### Step 2 – Adding your repo to the docs.json
 
 File a Pull Request within the [docs-page](https://github.com/SovereignCloudStack/docs-page) repository and add your repo to the docs.package.json:
 
 ```json5
```
Comment (joshmue) [inline community/contribute/adding-docs-guide.md:36]: As you dropped the comments from here to provide valid (plain) JSON, I think my change regarding JSON5 can be reverted:
```suggestion
```json
```

File: community/contribute/adding-docs-guide.md [hunk 2]
```diff
@@ -23,34 +23,37 @@ Your repository containing the documentation has to...
...
-        "repo": "demo-organisation/demo-repository",
-        // directory which shall be copied. Optional to specify only copying markdown files
-        "source": "doc/*.md",
-        // directory where the files should be copied to within the docs-page repo
-        "target": "docs",
-        // label for directory. only mandatory if source file is set to copy only *.md files and not the complete directory
-        "label": "demo-repository-label"
-    }
+  {
+    repo: "demo-organisation/demo-repository",
```
Comment (joshmue) [inline community/contribute/adding-docs-guide.md:39]: I think having keys without `"` only works in JS (and JSON5, YAML...), not in JSON.
```suggestion
    "repo": "demo-organisation/demo-repository",
```

File: community/contribute/adding-docs-guide.md [hunk 3]
```diff
@@ -23,34 +23,37 @@ Your repository containing the documentation has to...
...
+    target: "docs",
+    label: "demo-repository-label",
+  },
 ]
 ```
 
-Once it is approved and merged, a postinstall script will be triggered within the build process. This initiates downloading, copying and distilling which results in this static generated [documentation](https://docs.scs.community) page – now with your content.
+| key    | description                                                                                                       |
+| ------ | ----------------------------------------------------------------------------------------------------------------- |
+| repo   | link to github organisation and repository                                                                        |
```
Comment (joshmue) [inline community/contribute/adding-docs-guide.md:49]:
```suggestion
| repo   | reference to github organisation and repository                                                                   |
```

File: community/contribute/adding-docs-guide.md [hunk 4]
```diff
@@ -23,34 +23,37 @@ Your repository containing the documentation has to...
...
+| source | directory which shall be copied. Optional to specify only copying markdown files                                  |
```
Comment (joshmue) [inline community/contribute/adding-docs-guide.md:50]:
```suggestion
| source | path to content to copy: Either glob matching individual markdown files OR the path of a single directory         |
```

**Review by joshmue — APPROVED** [review]
Added some JSON/field-description nitpicking.

Overall: LGTM.

**Review by ra-beer — APPROVED** [review]
LGTM

**Comment by maxwolfs** [comment]
Thank you all!

**Pull request merged**